### PR TITLE
global integration test related fixes

### PR DESCRIPTION
**Pull request by jmozah**
- Removed chunk type test in `IsPotential` as this is unnecessary
- Removed chunk type from swarm.Chunk as it is not needed now 
- Added local existence for chunk when a request for repair is received, otherwise there will be unnecessary network Get (which will fail since someone triggered repair because the chunk is not there in network)

### CODE REVIEW

File: pkg/recovery/repair.go [hunk 1]
```diff
@@ -48,6 +48,19 @@ func NewRepairHandler(s storage.Storer, logger logging.Logger, pushSyncer pushsy
 	return func(ctx context.Context, m *trojan.Message) error {
 		chAddr := m.Payload
 
+		// check if the chunk exists in the local store and proceed.
+		// otherwise the Get will trigger a unnecessary network retrieve
+		exists, err := s.Has(ctx, swarm.NewAddress(chAddr))
+		if err != nil {
+			logger.Info("chunk repair: error while getting chunk for repairing: ", err)
+			return err
```
Comment (janos) [inline pkg/recovery/repair.go:56]: Why logging an error and returning it?

File: pkg/recovery/repair.go [hunk 2]
```diff
@@ -48,6 +48,19 @@ func NewRepairHandler(s storage.Storer, logger logging.Logger, pushSyncer pushsy
...
+		}
+		if !exists {
+			logger.Info("chunk repair: chunk not present in local store for repairing")
```
Comment (janos) [inline pkg/recovery/repair.go:59]: Why logging an error and returning it?

File: pkg/recovery/repair.go (hunk 1 above)
Comment (jmozah) [reply pkg/recovery/repair.go:56]: Creaped in from debugging session.. removing it

File: pkg/recovery/repair.go (hunk 2 above)
Comment (jmozah) [reply pkg/recovery/repair.go:59]: Creaped in from debugging session.. removing it

File: pkg/recovery/repair_test.go [hunk 3]
```diff
@@ -201,7 +201,7 @@ func TestNewRepairHandler(t *testing.T) {
 
 		// invoke the chunk repair handler
 		err = repairHandler(context.Background(), &msg)
-		if err != nil && err.Error() != "storage: not found" {
+		if err != nil && err.Error() != "chunk repair: chunk not present in local store for repairing" {
```
Comment (acud) [inline pkg/recovery/repair_test.go:204]: you can use `errors.Is()` if the error is extracted and exposed through `export_test`

File: pkg/recovery/repair.go [hunk 4]
```diff
@@ -48,6 +49,17 @@ func NewRepairHandler(s storage.Storer, logger logging.Logger, pushSyncer pushsy
...
 		chAddr := m.Payload
 
+		// check if the chunk exists in the local store and proceed.
+		// otherwise the Get will trigger a unnecessary network retrieve
+		exists, err := s.Has(ctx, swarm.NewAddress(chAddr))
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return fmt.Errorf("chunk repair: chunk not present in local store for repairing")
```
Comment (acud) [inline pkg/recovery/repair.go:59]: maybe would be nice to extract the error to be a `const` or `var` at the beginning of the file?

**Review by acud — APPROVED** [review]
minor comments. otherwise LGTM

File: pkg/recovery/repair_test.go (hunk 3 above)
Comment (janos) [reply pkg/recovery/repair_test.go:204]: Yes, correct, lets have a proper error checking, not string validation.

Comment (jmozah) [reply pkg/recovery/repair_test.go]: oh yeah.. sure

**Review by janos — APPROVED** [review]
Nice, thanks for addressing my comments!

**Pull request merged**